src/go/rpk/pkg/config/manager.go

```diff
@@ -382,6 +384,14 @@ func (m *manager) Set(key, value, format string) error {
 	return m.v.MergeConfigMap(newV.AllSettings())
 }
 
+func (m *manager) MergeConfigToCurrentMap(conf *Config) error {
```

Comment:
The map is the internal representation of the config, which I don't think we need to hint to in the name. `Merge` would be ok, I think and it's less verbose.

Reply:
Also please add some tests 🙏🏽

Reply:
> Merge would be ok, I think and it's less verbose.
sounds. good, I changed the name to `Merge`

I added a test for `Merge` and one case for `Other`. Let me know if that's what you had in mind.